packages/components/inputs/localized-money-input/src/localized-money-input.js

```diff
@@ -157,83 +143,66 @@ const LocalizedMoneyInput = (props) => {
+    currencies.length > 1 && !props.hideCurrencyExpansionControls;
+
   return (
     <Constraints.Horizontal constraint={props.horizontalConstraint}>
-      <Stack scale="s">
-        {currencies.map((currency, index) => {
-          const isFirstCurrency = index === 0;
-          if (!isFirstCurrency && !areCurrenciesExpanded) return null;
-          const isLastCurrency = index === currencies.length - 1;
-          const hasRemainingCurrencies = currencies.length > 1;
-          const hasErrorOnRemainingCurrencies =
-            props.hasError ||
-            getHasErrorOnRemainingLanguages(
-              props.errors,
-              props.selectedCurrency
-            );
-          const hasWarningOnRemainingCurrencies =
-            props.hasWarning ||
-            getHasWarningOnRemainingLanguages(
-              props.warnings,
-              props.selectedCurrency
+      <Stack scale="xs">
+        <Stack scale="s">
+          {currencies.map((currency, index) => {
+            const isFirstCurrency = index === 0;
+            if (!isFirstCurrency && !areCurrenciesExpanded) return null;
+
+            return (
+              <LocalizedInput
+                key={currency}
+                id={LocalizedMoneyInput.getId(id, currency)}
+                name={LocalizedMoneyInput.getName(props.name, currency)}
+                value={props.value[currency]}
+                onChange={props.onChange}
+                currency={currency}
+                placeholder={
+                  props.placeholder ? props.placeholder[currency] : undefined
+                }
+                onBlur={props.onBlur}
+                onFocus={props.onFocus}
+                isDisabled={props.isDisabled}
+                isReadOnly={props.isReadOnly}
+                hasError={Boolean(
+                  props.hasError || (props.errors && props.errors[currency])
+                )}
+                hasWarning={Boolean(
+                  props.hasWarning ||
+                    (props.warnings && props.warnings[currency])
+                )}
+                intl={intl}
+                warning={props.warnings && props.warnings[currency]}
+                error={props.errors && props.errors[currency]}
+                {...createLocalizedDataAttributes(props, currency)}
+              />
             );
-          return (
-            <LocalizedInput
-              key={currency}
-              id={LocalizedMoneyInput.getId(id, currency)}
-              name={LocalizedMoneyInput.getName(props.name, currency)}
-              value={props.value[currency]}
-              onChange={props.onChange}
-              currency={currency}
-              placeholder={
-                props.placeholder ? props.placeholder[currency] : undefined
-              }
-              onBlur={props.onBlur}
-              onFocus={props.onFocus}
-              isDisabled={props.isDisabled}
-              isReadOnly={props.isReadOnly}
-              currenciesControl={(() => {
```

Comment:
No need for this anymore, as well as the flags declared above within this .map function.